pkg/binlog/position_test.go

```diff
@@ -723,3 +723,45 @@ func (t *testPositionSuite) TestVerifyBinlogPos(c *C) {
 		}
 	}
 }
+
+func (t *testPositionSuite) TestSetGTID(c *C) {
+	GTIDSetStr := "3ccc475b-2343-11e7-be21-6c0b84d59f30:1-14"
+	GTIDSetStr2 := "3ccc475b-2343-11e7-be21-6c0b84d59f30:1-15"
+	gset, _ := gtid.ParserGTID("mysql", GTIDSetStr)
+	gset2, _ := gtid.ParserGTID("mysql", GTIDSetStr2)
+	mysqlSet := gset.Origin()
+	mysqlSet2 := gset2.Origin()
+
+	loc := Location{
+		Position: gmysql.Position{
+			Name: "mysql-bin.00002",
+			Pos:  2333,
+		},
+		GTIDSet: gset,
+		Suffix:  0,
+	}
+	loc2 := loc
+
+	c.Assert(CompareLocation(loc, loc2, false), Equals, 0)
+
+	loc2.Position.Pos++
+	c.Assert(loc.Position.Pos, Equals, uint32(2333))
+	c.Assert(CompareLocation(loc, loc2, false), Equals, -1)
+
+	loc2.Position.Name = "mysql-bin.00001"
+	c.Assert(loc.Position.Name, Equals, "mysql-bin.00002")
+	c.Assert(CompareLocation(loc, loc2, false), Equals, 1)
+
+	// WARN: will change other location's gtid
+	err := loc2.GTIDSet.Set(mysqlSet2)
```

Comment:
agree to make member `GTIDSet` private, to avoid this usage (if need much change, later is fine)